plugin/src/main/java/io/jenkins/plugins/analysis/core/model/ColumnMarker.java

```diff
@@ -0,0 +1,47 @@
+package io.jenkins.plugins.analysis.core.model;
+
+/**
+ * Encloses columns between start and end with the HTML tag 'mark'.
+ *
+ */
+public final class ColumnMarker {
+    private ColumnMarker() {
+        throw new UnsupportedOperationException();
+    }
+    /**
+     * Encloses columns between start and end with the HTML tag 'mark'.
+     * This will make prism highlight the enclosed part of the line.
+     *
+     * @param text
+     *         the source code line
+     * @param start
+     *         the first column in text, that needs to be marked
+     * @param end
+     *         the last column in text, that needs to be marked
+     *
+     * @return StringBuilder containing the text with the added html tag "mark"
+     */
+    public static StringBuilder markColumns(final String text, final int start, final int end) {
+        if (start < 1 || text.length() == 0 || end > text.length()) {
+            return new StringBuilder(text);
+        }
+        final int realStart = start - 1;
+        final int realEnd = (end == 0) ? text.length() - 1 : end - 1;
+        final int afterMark = realEnd + 1;
+
+        if (realStart > realEnd) {
+            return new StringBuilder(text);
+        }
+
+        final String before = text.substring(0, realStart);
+        final String toBeMarked = text.substring(realStart, afterMark);
+        final String after = text.substring(afterMark);
+
+        return new StringBuilder(before)
+                .append("<mark>")
+                .append(toBeMarked)
+                .append("</mark>")
+                .append(after);
+    }
```

Comment:
I got it working locally now :tada: I needed to disable the sanitizer that ensures that no one uses a source code file to run JS code. I think we need to add the marks right after the Sanitzer finished the processing of the source code.

Reply:
This part: https://github.com/jenkinsci/warnings-ng-plugin/blob/91ca68b00b7e10e95b29a7bc213b35f18ec3b0ed/plugin/src/main/java/io/jenkins/plugins/analysis/core/model/SourcePrinter.java#L135
We need to return the text, inject the marks, and wrap it into an `UnescapedBlock`.

Reply:
I'm not sure if with actual source code the text changes so that columns are at different positions. This is something we should test...

Reply:
if I do it like :
~~~~ .java
    private String asMarkedCode(final StringBuilder text, Issue issue, final String... classes) {
        final String sanitized = SANITIZER.render(StringEscapeUtils.escapeHtml4(text.toString()));
        final StringBuilder marked = ColumnMarker.markColumns(sanitized, issue.getColumnStart(), issue.getColumnEnd());
        final UnescapedText unescapedText = new UnescapedText(marked.toString());
        return code().withClasses(classes).with(unescapedText).render();
    }
~~~~
The string `sanitized` contains: `  int b &#61; std::move(argc);`.
that makes the marking fail.

Could we make the sanitizer ignoring `<mark></mark>`?

Reply:
This is a good idea. I don't think that we can ignore `mark` but we can use something different (e.g. `<b>`) and replace that after the Sanitizer worked on the block.

Reply:
I was about  writing some similar:

Instead of marking the text with an html tag, it should be a text that is unlikely to be used in any source code.
The replace it after sanitizer like you wrote.

Reply:
Yes, that should work as well!